uhubctl.c

```diff
@@ -507,6 +511,94 @@ static int get_port_status(struct libusb_device_handle *devh, int port)
 }
 
 
+#ifdef __gnu_linux__
+/*
+ * Try to use the Linux sysfs interface to power a port off/on.
+ * Returns 0 on success.
+ */
+
+static int set_port_status_linux(struct libusb_device_handle *devh, struct hub_info *hub, int port, int on)
+{
+    int configuration = 0;
+    char disable_path[PATH_MAX];
+
+    int rc = libusb_get_configuration(devh, &configuration);
+    if (rc < 0) {
+        return rc;
+    }
+
+    // The "disable" sysfs interface is available starting with kernel version
+    // 6.0, for earlier kernel versions the open() call will fail and we fall
+    // back to using libusb.
+    snprintf(disable_path, PATH_MAX,
+        "/sys/bus/usb/devices/%s:%d.0/%s-port%i/disable",
+        hub->location, configuration, hub->location, port
+    );
+
+    int disable_fd = open(disable_path, O_WRONLY);
```

Comment:
If user is not using sudo, this path exists but cannot be opened for writing, perhaps it should complain here about permissions?
E.g. we can use `access()` to check if file exists. If it does not, we fall back to libusb interface silently.
But if disable file exists and we cannot open it for writing, user should be aware that he is doing something wrong.